components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/APIExecutor.java

```diff
@@ -178,7 +178,7 @@ private boolean changeLifeCycle(RequestContext context, API api, Resource apiRes
                 String apiSecurity = api.getApiSecurity();
                 boolean isOauthProtected =
                         apiSecurity == null || apiSecurity.contains(APIConstants.DEFAULT_API_SECURITY_OAUTH2);
-                if (endPoint != null && endPoint.trim().length() > 0) {
+                if (api.getType().equals(APIConstants.API_TYPE_WEBSUB) || (endPoint != null && endPoint.trim().length() > 0)) {
```

Comment:
Better to use the equals method the other way round to avoid NPE

Reply:
+1